rstest_macros/src/render/mod.rs

```diff
@@ -260,6 +280,19 @@ fn single_test_case<'a>(
     };
     let execute = render_test_call(testfn_name.clone().into(), &args, timeout, is_async);
 
+    let test_impl = test_impl.map(|test_impl| {
+        let mut test_impl = test_impl.clone();
+        for arg in test_impl.sig.inputs.iter_mut() {
+            if let FnArg::Typed(t) = arg {
+                t.attrs = std::mem::take(&mut t.attrs)
+                    .into_iter()
+                    .filter(|a| !attr_is(a, "future"))
+                    .collect();
+            }
+        }
+        test_impl
+    });
+
     quote! {
```

Comment:
Why you're removing the `#[future]` attrs?

Reply:
Whoops, I think that's code that I left over from when I was hacking the solution, I can remove it now